packages/rich-text/src/component/index.native.js

```diff
@@ -20,6 +20,7 @@ import { childrenBlock } from '@wordpress/blocks';
 import { decodeEntities } from '@wordpress/html-entities';
 import { BACKSPACE } from '@wordpress/keycodes';
 import { isURL } from '@wordpress/url';
+import { withTheme } from '@wordpress/components';
```

Comment:
That's interesting. It's not part of the web public API. Do you think it would be helpful to start using the wrapper components for theming in general?

Reply:
I wouldn't port this HOC to the web, at least not in its current state. It uses an internal event emitter to listen to native events and refresh the styles and is very much tied to the use case of dark mode right now. I think we will want to refactor this even for native first and use the store to dispatch events at least.

Reply:
Sounds like `__experimentalWithDarkMode` then :)